jsonrpc2.go

```diff
@@ -439,13 +439,11 @@ func (c *Conn) readMessages(ctx context.Context) {
 	var err error
 	for err == nil {
 		var m anyMessage
-
-		var objBytes []byte
-		objBytes, err = c.transport.ReadObject()
-		if err != nil {
-			break
+		var r io.Reader
+		r, err = c.transport.NextObjectReader()
+		if err == nil {
+			err = json.NewDecoder(r).Decode(&m)
```

Comment:
ok my earlier comments are somewhat addressed here, but I'll mention the other part here. Should we be using a stream decoder here? This will nearly always be a limit reader so we can do better.
Also creating a NewDecoder each time just hides the underlying `[]byte` we create each time (it lives in the struct returned by `json.NewDecoder`).